src/librustc_metadata/rmeta/encoder.rs

```diff
@@ -1580,6 +1586,12 @@ impl EncodeContext<'tcx> {
         }
     }
 
+    fn encode_ident_span(&mut self, def_id: DefId) {
+        if let Some(ident_span) = self.tcx.hir().definitions().ident_span(def_id.index) {
```

Comment:
Why is the intermediate table in `struct Definitions` necessary?
All the required idents seem to be available at `encode_ident_span` call sites.

Reply:
I thought that the idents weren't available during metadata encoding, but it turns out that they are. I've updated the PR.